{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Main.Kraft"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "include(\"../src/Kraft.jl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking programs ...\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mskewer\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/skewer\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mfastqc\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/fastqc\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mbgzip\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/bgzip\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mtabix\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/tabix\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mminimap2\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/minimap2\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4msamtools\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/samtools\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mbcftools\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/bcftools\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4msnpEff\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/snpEff\n",
      "`\u001b[4mwhich\u001b[24m \u001b[4mkallisto\u001b[24m`\n",
      "/home/kwat/miniconda3/bin/kallisto\n",
      "`\u001b[4mbash\u001b[24m \u001b[4m-c\u001b[24m \u001b[4m'source activate py2.7 && which configManta.py'\u001b[24m`\n",
      "/home/kwat/miniconda3/envs/py2.7/bin/configManta.py\n",
      "`\u001b[4mbash\u001b[24m \u001b[4m-c\u001b[24m \u001b[4m'source activate py2.7 && which configureStrelkaGermlineWorkflow.py'\u001b[24m`\n",
      "/home/kwat/miniconda3/envs/py2.7/bin/configureStrelkaGermlineWorkflow.py\n",
      "`\u001b[4mbash\u001b[24m \u001b[4m-c\u001b[24m \u001b[4m'source activate py2.7 && which configureStrelkaSomaticWorkflow.py'\u001b[24m`\n",
      "/home/kwat/miniconda3/envs/py2.7/bin/configureStrelkaSomaticWorkflow.py\n"
     ]
    }
   ],
   "source": [
    "Kraft.check_programs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making path dict ...\n",
      "/home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "/home/kwat/garden/data/grch/chromosome.bed.gz\n",
      "/home/kwat/garden/data/grch/chrn_n.tsv\n",
      "/home/kwat/garden/data/grch/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
      "/home/kwat/garden/data/grch/enst_gene_name.tsv\n",
      "/home/kwat/garden/data/virus/sequences.fasta\n",
      "/home/kwat/garden/data/virus/sequences.fasta\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Dict{String,String} with 6 entries:\n",
       "  \"dna.fa.gz\"         => \"/home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_n…\n",
       "  \"chrn_n.tsv\"        => \"/home/kwat/garden/data/grch/chrn_n.tsv\"\n",
       "  \"chromosome.bed.gz\" => \"/home/kwat/garden/data/grch/chromosome.bed.gz\"\n",
       "  \"virus_cdna.fa.gz\"  => \"/home/kwat/garden/data/virus/sequences.fasta\"\n",
       "  \"cdna.fa.gz\"        => \"/home/kwat/garden/data/grch/Homo_sapiens.GRCh38.cdna.…\n",
       "  \"dna.fa.bgz\"        => \"/home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_n…"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "const path_dict = Kraft.make_path_dict(\"/home/kwat/garden/data\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "const germ_dna_1_fq_gz = \"/home/kwat/garden/sample/germ_dna.1.fastq.gz\"\n",
    "\n",
    "const germ_dna_2_fq_gz = \"/home/kwat/garden/sample/germ_dna.2.fastq.gz\"\n",
    "\n",
    "const is_targeted = true\n",
    "\n",
    "const output_dir = \"/home/kwat/garden/sample/process_germ_dna\"\n",
    "\n",
    "const n_job = 2\n",
    "\n",
    "const gb_memory = 16\n",
    "\n",
    "const job_gb_memory = 1;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file_path::String in(\n",
    "    germ_dna_1_fq_gz,\n",
    "    germ_dna_2_fq_gz,\n",
    ")\n",
    "    \n",
    "    if !isfile(file_path)\n",
    "        \n",
    "        error(\"$file_path doesn't exist.\")\n",
    "    \n",
    "    end\n",
    "    \n",
    "end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2019-06-24T21:21:35.013) Trimming sequence ...\n",
      "`\u001b[4mskewer\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m-x\u001b[24m \u001b[4mAGATCGGAAGAGC\u001b[24m \u001b[4m--compress\u001b[24m \u001b[4m--output\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ\u001b[24m \u001b[4m--quiet\u001b[24m \u001b[4m/home/kwat/garden/sample/germ_dna.1.fastq.gz\u001b[24m \u001b[4m/home/kwat/garden/sample/germ_dna.2.fastq.gz\u001b[24m`\n",
      ".--. .-.\n",
      ": .--': :.-.\n",
      "`. `. : `'.' .--. .-..-..-. .--. .--.\n",
      "_`, :: . `.' '_.': `; `; :' '_.': ..'\n",
      "`.__.':_;:_;`.__.'`.__.__.'`.__.':_;\n",
      "skewer v0.2.2 [April 4, 2016]\n",
      "Parameters used:\n",
      "-- 3' end adapter sequence (-x):\u001b[0;33m\tAGATCGGAAGAGC\n",
      "\u001b[0m-- maximum error ratio allowed (-r):\t0.100\n",
      "-- maximum indel error ratio allowed (-d):\t0.030\n",
      "-- minimum read length allowed after trimming (-l):\t18\n",
      "-- file format (-f):\t\tSanger/Illumina 1.8+ FASTQ (auto detected)\n",
      "-- number of concurrent threads (-t):\t2\n",
      "Mon Jun 24 21:21:35 2019\u001b[0;32m >> started\u001b[0m\n",
      "\n",
      "Mon Jun 24 21:21:56 2019\u001b[0;32m >> done\u001b[0m (21.252s)\n",
      "1000000 read pairs processed; of these:\n",
      "      9 ( 0.00%) short read pairs filtered out after trimming by size control\n",
      "      1 ( 0.00%) empty read pairs filtered out after trimming by size control\n",
      " 999990 (100.00%) read pairs available; of these:\n",
      "    506 ( 0.05%) trimmed read pairs available after processing\n",
      " 999484 (99.95%) untrimmed read pairs available after processing\n",
      "log has been saved to \"/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed.log\".\n",
      "(2019-06-24T21:21:56.291) Done in 21 seconds, 278 milliseconds.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair2.fastq.gz\""
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "const germ_trim_sequence_prefix = joinpath(\n",
    "    output_dir,\n",
    "    \"trim_sequence\",\n",
    "    \"germ\",\n",
    ")\n",
    "\n",
    "Kraft.trim_sequence(\n",
    "    germ_dna_1_fq_gz,\n",
    "    germ_dna_2_fq_gz,\n",
    "    germ_trim_sequence_prefix,\n",
    "    n_job,\n",
    ")\n",
    "\n",
    "const germ_trim_1_fq_gz = \"$germ_trim_sequence_prefix-trimmed-pair1.fastq.gz\"\n",
    "\n",
    "const germ_trim_2_fq_gz = replace(\n",
    "    germ_trim_1_fq_gz,\n",
    "    \"pair1\"=>\"pair2\",\n",
    ");"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2019-06-24T21:21:56.626) Checking sequence ...\n",
      "`\u001b[4mfastqc\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m--outdir\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/check_sequence\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair1.fastq.gz\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair2.fastq.gz\u001b[24m`\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of germ-trimmed-pair1.fastq.gz\n",
      "Started analysis of germ-trimmed-pair2.fastq.gz\n",
      "Approx 5% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 5% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 10% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 10% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 15% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 15% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 20% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 20% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 25% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 25% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 30% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 30% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 35% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 35% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 40% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 40% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 45% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 45% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 50% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 50% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 55% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 55% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 60% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 60% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 65% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 65% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 70% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 70% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 75% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 75% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 80% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 80% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 85% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 85% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 90% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 90% complete for germ-trimmed-pair2.fastq.gz\n",
      "Approx 95% complete for germ-trimmed-pair1.fastq.gz\n",
      "Approx 95% complete for germ-trimmed-pair2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for germ-trimmed-pair1.fastq.gz\n",
      "Analysis complete for germ-trimmed-pair2.fastq.gz\n",
      "(2019-06-24T21:22:09.975) Done in 13 seconds, 349 milliseconds.\n"
     ]
    }
   ],
   "source": [
    "Kraft.check_sequence(\n",
    "    (\n",
    "        germ_trim_1_fq_gz,\n",
    "        germ_trim_2_fq_gz,\n",
    "    ),\n",
    "    joinpath(\n",
    "        output_dir,\n",
    "        \"check_sequence\",\n",
    "    ),\n",
    "    n_job,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2019-06-24T21:22:10.107) Aligning sequence ...\n",
      "pipeline(pipeline(pipeline(pipeline(`\u001b[4mminimap2\u001b[24m \u001b[4m-x\u001b[24m \u001b[4msr\u001b[24m \u001b[4m-t\u001b[24m \u001b[4m2\u001b[24m \u001b[4m-K\u001b[24m \u001b[4m1G\u001b[24m \u001b[4m-R\u001b[24m \u001b[4m'@RG\\tID:Germ\\tSM:Germ'\u001b[24m \u001b[4m-a\u001b[24m \u001b[4m/home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz.mmi\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair1.fastq.gz\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair2.fastq.gz\u001b[24m`, stdout=`\u001b[4msamtools\u001b[24m \u001b[4msort\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m-m\u001b[24m \u001b[4m1G\u001b[24m \u001b[4m-n\u001b[24m`), stdout=`\u001b[4msamtools\u001b[24m \u001b[4mfixmate\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m-m\u001b[24m \u001b[4m-\u001b[24m \u001b[4m-\u001b[24m`), stdout=`\u001b[4msamtools\u001b[24m \u001b[4msort\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m-m\u001b[24m \u001b[4m1G\u001b[24m`), stdout=Base.FileRedirect(\"/home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam.tmp\", false))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WARNING]\u001b[1;31m Indexing parameters (-k, -w or -H) overridden by parameters used in the prebuilt index.\u001b[0m\n",
      "[M::main::7.942*0.90] loaded/built the index for 195 target sequence(s)\n",
      "[M::mm_mapopt_update::7.942*0.90] mid_occ = 1000\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 195\n",
      "[M::mm_idx_stat::8.802*0.91] distinct minimizers: 100167746 (38.80% are singletons); average occurrences: 5.519; average spacing: 5.607\n",
      "[M::worker_pipeline::97.186*1.83] mapped 1999980 sequences\n",
      "[M::main] Version: 2.17-r941\n",
      "[M::main] CMD: minimap2 -x sr -t 2 -K 1G -R @RG\\tID:Germ\\tSM:Germ -a /home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz.mmi /home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair1.fastq.gz /home/kwat/garden/sample/process_germ_dna/trim_sequence/germ-trimmed-pair2.fastq.gz\n",
      "[M::main] Real time: 97.443 sec; CPU: 178.011 sec; Peak RSS: 8.534 GB\n",
      "[bam_sort_core] merging from 0 files and 2 in-memory blocks...\n",
      "[bam_sort_core] merging from 0 files and 2 in-memory blocks...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "`\u001b[4msamtools\u001b[24m \u001b[4mmarkdup\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m-s\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam.tmp\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam\u001b[24m`\n",
      "`\u001b[4msamtools\u001b[24m \u001b[4mindex\u001b[24m \u001b[4m-@\u001b[24m \u001b[4m2\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam\u001b[24m`\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "READ 2002462 WRITTEN 2002462 \n",
      "EXCLUDED 8602 EXAMINED 1993860\n",
      "PAIRED 1988572 SINGLE 5288\n",
      "DULPICATE PAIR 57366 DUPLICATE SINGLE 1279\n",
      "DUPLICATE TOTAL 58645\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pipeline(`\u001b[4msamtools\u001b[24m \u001b[4mflagstat\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam\u001b[24m`, stdout=Base.FileRedirect(\"/home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam.flagstat\", false))\n",
      "(2019-06-24T21:24:07.375) Done in 1 minute, 57 seconds, 268 milliseconds.\n"
     ]
    }
   ],
   "source": [
    "const germ_bam = joinpath(\n",
    "    output_dir,\n",
    "    \"align_sequence\",\n",
    "    \"germ_dna.bam\",\n",
    ")\n",
    "\n",
    "Kraft.align_sequence(\n",
    "    germ_trim_1_fq_gz,\n",
    "    germ_trim_2_fq_gz,\n",
    "    \"Germ\",\n",
    "    path_dict[\"dna.fa.gz\"],\n",
    "    germ_bam,\n",
    "    n_job,\n",
    "    job_gb_memory,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2019-06-24T21:24:07.524) Finding variant ...\n",
      "`\u001b[4mbash\u001b[24m \u001b[4m-c\u001b[24m \u001b[4m'source activate py2.7 && configManta.py --referenceFasta /home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bgz --callRegions /home/kwat/garden/data/grch/chromosome.bed.gz --exome --bam /home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam --outputContig --runDir /home/kwat/garden/sample/process_germ_dna/find_variant/manta && /home/kwat/garden/sample/process_germ_dna/find_variant/manta/runWorkflow.py --mode local --jobs 2 --memGb 16 --quiet'\u001b[24m`\n",
      "\n",
      "Successfully created workflow run script.\n",
      "To execute the workflow, run the following script and set appropriate options:\n",
      "\n",
      "/home/kwat/garden/sample/process_germ_dna/find_variant/manta/runWorkflow.py\n",
      "`\u001b[4mbash\u001b[24m \u001b[4m-c\u001b[24m \u001b[4m'source activate py2.7 && configureStrelkaGermlineWorkflow.py --referenceFasta /home/kwat/garden/data/grch/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bgz --callRegions /home/kwat/garden/data/grch/chromosome.bed.gz --exome --bam /home/kwat/garden/sample/process_germ_dna/align_sequence/germ_dna.bam --runDir /home/kwat/garden/sample/process_germ_dna/find_variant/strelka && /home/kwat/garden/sample/process_germ_dna/find_variant/strelka/runWorkflow.py --mode local --jobs 2 --memGb 16 --quiet'\u001b[24m`\n",
      "\n",
      "Successfully created workflow run script.\n",
      "To execute the workflow, run the following script and set appropriate options:\n",
      "\n",
      "/home/kwat/garden/sample/process_germ_dna/find_variant/strelka/runWorkflow.py\n",
      "pipeline(pipeline(pipeline(`\u001b[4mbcftools\u001b[24m \u001b[4mconcat\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m--allow-overlaps\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/manta/results/variants/diploidSV.vcf.gz\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/strelka/results/variants/variants.vcf.gz\u001b[24m`, stdout=`\u001b[4mbcftools\u001b[24m \u001b[4mannotate\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m--rename-chrs\u001b[24m \u001b[4m/home/kwat/garden/data/grch/chrn_n.tsv\u001b[24m`), stdout=`\u001b[4mbgzip\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m--stdout\u001b[24m`), stdout=Base.FileRedirect(\"/home/kwat/garden/sample/process_germ_dna/find_variant/concat.vcf.gz\", false))\n",
      "`\u001b[4mtabix\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/concat.vcf.gz\u001b[24m`\n",
      "pipeline(pipeline(`\u001b[4msnpEff\u001b[24m \u001b[4m-Xmx16g\u001b[24m \u001b[4mGRCh38.86\u001b[24m \u001b[4m-noLog\u001b[24m \u001b[4m-verbose\u001b[24m \u001b[4m-csvStats\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/stats.csv\u001b[24m \u001b[4m-htmlStats\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/stats.html\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/concat.vcf.gz\u001b[24m`, stdout=`\u001b[4mbgzip\u001b[24m \u001b[4m--threads\u001b[24m \u001b[4m2\u001b[24m \u001b[4m--stdout\u001b[24m`), stdout=Base.FileRedirect(\"/home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/snpeff.vcf.gz\", false))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "00:00:00\tSnpEff version SnpEff 4.3t (build 2017-11-24 10:18), by Pablo Cingolani\n",
      "00:00:00\tCommand: 'ann'\n",
      "00:00:00\tReading configuration file 'snpEff.config'. Genome: 'GRCh38.86'\n",
      "00:00:00\tReading config file: /home/kwat/github/Kraft.jl/notebook/snpEff.config\n",
      "00:00:00\tReading config file: /home/kwat/miniconda3/share/snpeff-4.3.1t-2/snpEff.config\n",
      "00:00:00\tdone\n",
      "00:00:00\tReading database for genome version 'GRCh38.86' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/snpEffectPredictor.bin' (this might take a while)\n",
      "00:00:13\tdone\n",
      "00:00:13\tReading NextProt database from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/nextProt.bin'\n",
      "00:00:14\tNextProt database: 652574 markers loaded.\n",
      "00:00:14\tAdding transcript info to NextProt markers.\n",
      "00:00:14\tNextProt database: 652574 markers added.\n",
      "00:00:14\tLoading PWMs from : /home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/pwms.bin\n",
      "00:00:14\tLoading Motifs from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/motif.bin'\n",
      "00:00:14\t\tMotif database: 511011 markers loaded.\n",
      "00:00:15\tLoading interactions from : /home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/interactions.bin\n",
      "00:00:21\t\tInteractions: 1793688 added, 0 skipped.\n",
      "00:00:21\tBuilding interval forest\n",
      "00:00:26\tdone.\n",
      "00:00:26\tGenome stats :\n",
      "#-----------------------------------------------\n",
      "# Genome name                : 'Homo_sapiens'\n",
      "# Genome version             : 'GRCh38.86'\n",
      "# Genome ID                  : 'GRCh38.86[0]'\n",
      "# Has protein coding info    : true\n",
      "# Has Tr. Support Level info : true\n",
      "# Genes                      : 58051\n",
      "# Protein coding genes       : 20423\n",
      "#-----------------------------------------------\n",
      "# Transcripts                : 198002\n",
      "# Avg. transcripts per gene  : 3.41\n",
      "# TSL transcripts            : 166906\n",
      "#-----------------------------------------------\n",
      "# Checked transcripts        : \n",
      "#               AA sequences :      0 ( 0.00% )\n",
      "#              DNA sequences : 163038 ( 82.34% )\n",
      "#-----------------------------------------------\n",
      "# Protein coding transcripts : 94384\n",
      "#              Length errors :  13357 ( 14.15% )\n",
      "#  STOP codons in CDS errors :     51 ( 0.05% )\n",
      "#         START codon errors :  11250 ( 11.92% )\n",
      "#        STOP codon warnings :   7169 ( 7.60% )\n",
      "#              UTR sequences :  91549 ( 46.24% )\n",
      "#               Total Errors :  23099 ( 24.47% )\n",
      "#-----------------------------------------------\n",
      "# Cds                        : 704604\n",
      "# Exons                      : 1182163\n",
      "# Exons with sequence        : 1182163\n",
      "# Exons without sequence     : 0\n",
      "# Avg. exons per transcript  : 5.97\n",
      "# WARNING!                   : Mitochondrion chromosome 'MT' does not have a mitochondrion codon table (codon table = 'Standard'). You should update the config file.\n",
      "#-----------------------------------------------\n",
      "# Number of chromosomes      : 524\n",
      "# Chromosomes                : Format 'chromo_name size codon_table'\n",
      "#\t\t'HSCHR1_2_CTG3'\t248975002\tStandard\n",
      "#\t\t'HSCHR1_1_CTG31'\t248973653\tStandard\n",
      "#\t\t'HSCHR1_2_CTG31'\t248971826\tStandard\n",
      "#\t\t'HSCHR1_1_CTG3'\t248968406\tStandard\n",
      "#\t\t'HSCHR1_3_CTG31'\t248967909\tStandard\n",
      "#\t\t'HSCHR1_4_CTG3'\t248963886\tStandard\n",
      "#\t\t'HSCHR1_2_CTG32_1'\t248962488\tStandard\n",
      "#\t\t'HSCHR1_3_CTG3'\t248961700\tStandard\n",
      "#\t\t'HSCHR1_1_CTG11'\t248961539\tStandard\n",
      "#\t\t'HSCHR1_3_CTG32_1'\t248960761\tStandard\n",
      "#\t\t'HSCHR1_5_CTG32_1'\t248960213\tStandard\n",
      "#\t\t'HSCHR1_1_CTG32_1'\t248960143\tStandard\n",
      "#\t\t'HG1832_PATCH'\t248959779\tStandard\n",
      "#\t\t'HG2095_PATCH'\t248959131\tStandard\n",
      "#\t\t'HSCHR1_ALT2_1_CTG32_1'\t248958866\tStandard\n",
      "#\t\t'HSCHR1_4_CTG32_1'\t248958766\tStandard\n",
      "#\t\t'HG2104_PATCH'\t248958259\tStandard\n",
      "#\t\t'HG986_PATCH'\t248956639\tStandard\n",
      "#\t\t'1'\t248956422\tStandard\n",
      "#\t\t'HG2058_PATCH'\t248956422\tStandard\n",
      "#\t\t'HSCHR1_4_CTG31'\t248955631\tStandard\n",
      "#\t\t'HG1342_HG2282_PATCH'\t248929249\tStandard\n",
      "#\t\t'HSCHR1_5_CTG3'\t248926662\tStandard\n",
      "#\t\t'HG2290_PATCH'\t242235269\tStandard\n",
      "#\t\t'HSCHR2_2_CTG7'\t242223512\tStandard\n",
      "#\t\t'HSCHR2_1_CTG7_2'\t242210518\tStandard\n",
      "#\t\t'HSCHR2_5_CTG7_2'\t242199685\tStandard\n",
      "#\t\t'HSCHR2_2_CTG7_2'\t242199609\tStandard\n",
      "#\t\t'HSCHR2_1_CTG5'\t242199562\tStandard\n",
      "#\t\t'HSCHR2_1_CTG15'\t242198904\tStandard\n",
      "#\t\t'HSCHR2_2_CTG15'\t242198544\tStandard\n",
      "#\t\t'HSCHR2_2_CTG1'\t242198346\tStandard\n",
      "#\t\t'HSCHR2_6_CTG7_2'\t242198149\tStandard\n",
      "#\t\t'HSCHR2_4_CTG7_2'\t242196589\tStandard\n",
      "#\t\t'HSCHR2_3_CTG1'\t242195626\tStandard\n",
      "#\t\t'HSCHR2_1_CTG1'\t242194896\tStandard\n",
      "#\t\t'HSCHR2_3_CTG7_2'\t242194557\tStandard\n",
      "#\t\t'HSCHR2_3_CTG15'\t242194292\tStandard\n",
      "#\t\t'HSCHR2_1_CTG7'\t242193540\tStandard\n",
      "#\t\t'HG2233_PATCH'\t242193535\tStandard\n",
      "#\t\t'2'\t242193529\tStandard\n",
      "#\t\t'HG2232_PATCH'\t242193529\tStandard\n",
      "#\t\t'HSCHR2_4_CTG1'\t242191076\tStandard\n",
      "#\t\t'HSCHR3_1_CTG2_1'\t198306097\tStandard\n",
      "#\t\t'HSCHR3_1_CTG3'\t198302882\tStandard\n",
      "#\t\t'HSCHR3_4_CTG2_1'\t198301383\tStandard\n",
      "#\t\t'HSCHR3_5_CTG2_1'\t198300594\tStandard\n",
      "#\t\t'HSCHR3_3_CTG2_1'\t198300421\tStandard\n",
      "#\t\t'HSCHR3_3_CTG1'\t198300110\tStandard\n",
      "#\t\t'HSCHR3_2_CTG2_1'\t198298736\tStandard\n",
      "#\t\t'HSCHR3_1_CTG1'\t198297480\tStandard\n",
      "#\t\t'HSCHR3_2_CTG3'\t198296961\tStandard\n",
      "#\t\t'HSCHR3_4_CTG3'\t198296867\tStandard\n",
      "#\t\t'HG2022_PATCH'\t198295941\tStandard\n",
      "#\t\t'HSCHR3_8_CTG3'\t198295934\tStandard\n",
      "#\t\t'HG126_PATCH'\t198295908\tStandard\n",
      "#\t\t'HG2235_PATCH'\t198295655\tStandard\n",
      "#\t\t'3'\t198295559\tStandard\n",
      "#\t\t'HG2066_PATCH'\t198295555\tStandard\n",
      "#\t\t'HSCHR3_6_CTG3'\t198294921\tStandard\n",
      "#\t\t'HSCHR3_7_CTG3'\t198294497\tStandard\n",
      "#\t\t'HSCHR3_5_CTG3'\t198293785\tStandard\n",
      "#\t\t'HSCHR3_3_CTG3'\t198293223\tStandard\n",
      "#\t\t'HSCHR3_9_CTG3'\t198291952\tStandard\n",
      "#\t\t'HG2237_PATCH'\t198290329\tStandard\n",
      "#\t\t'HSCHR4_11_CTG12'\t190233769\tStandard\n",
      "#\t\t'HSCHR4_1_CTG12'\t190227204\tStandard\n",
      "#\t\t'HSCHR4_7_CTG12'\t190225141\tStandard\n",
      "#\t\t'HSCHR4_1_CTG6'\t190224838\tStandard\n",
      "#\t\t'HSCHR4_1_CTG4'\t190221757\tStandard\n",
      "#\t\t'HSCHR4_4_CTG12'\t190220704\tStandard\n",
      "#\t\t'HSCHR4_5_CTG12'\t190219562\tStandard\n",
      "#\t\t'HSCHR4_9_CTG12'\t190219005\tStandard\n",
      "#\t\t'HSCHR4_1_CTG8_1'\t190218838\tStandard\n",
      "#\t\t'HSCHR4_6_CTG12'\t190218533\tStandard\n",
      "#\t\t'HSCHR4_8_CTG12'\t190218480\tStandard\n",
      "#\t\t'HSCHR4_3_CTG12'\t190218028\tStandard\n",
      "#\t\t'HSCHR4_2_CTG4'\t190217898\tStandard\n",
      "#\t\t'4'\t190214555\tStandard\n",
      "#\t\t'HSCHR4_2_CTG12'\t190206797\tStandard\n",
      "#\t\t'HG2023_PATCH'\t190147504\tStandard\n",
      "#\t\t'HSCHR4_1_CTG9'\t190096932\tStandard\n",
      "#\t\t'HSCHR5_1_CTG1'\t181558402\tStandard\n",
      "#\t\t'HSCHR5_3_CTG1_1'\t181554865\tStandard\n",
      "#\t\t'HSCHR5_2_CTG1'\t181547139\tStandard\n",
      "#\t\t'HSCHR5_4_CTG1_1'\t181546281\tStandard\n",
      "#\t\t'HSCHR5_7_CTG1'\t181544732\tStandard\n",
      "#\t\t'HSCHR5_1_CTG5'\t181543736\tStandard\n",
      "#\t\t'HSCHR5_3_CTG1'\t181542495\tStandard\n",
      "#\t\t'HSCHR5_2_CTG5'\t181542206\tStandard\n",
      "#\t\t'HSCHR5_6_CTG1'\t181540245\tStandard\n",
      "#\t\t'HSCHR5_5_CTG1'\t181539933\tStandard\n",
      "#\t\t'HSCHR5_3_CTG5'\t181538569\tStandard\n",
      "#\t\t'5'\t181538259\tStandard\n",
      "#\t\t'HSCHR5_4_CTG1'\t181537434\tStandard\n",
      "#\t\t'HSCHR5_1_CTG1_1'\t180863570\tStandard\n",
      "#\t\t'HSCHR5_2_CTG1_1'\t180753562\tStandard\n",
      "#\t\t'HSCHR6_MHC_MCF_CTG1'\t170976334\tStandard\n",
      "#\t\t'HSCHR6_MHC_MANN_CTG1'\t170959829\tStandard\n",
      "#\t\t'HSCHR6_MHC_SSTO_CTG1'\t170946136\tStandard\n",
      "#\t\t'HSCHR6_MHC_APD_CTG1'\t170845044\tStandard\n",
      "#\t\t'HSCHR6_1_CTG3'\t170836231\tStandard\n",
      "#\t\t'HSCHR6_1_CTG2'\t170833811\tStandard\n",
      "#\t\t'HSCHR6_1_CTG6'\t170818797\tStandard\n",
      "#\t\t'HG2072_PATCH'\t170816182\tStandard\n",
      "#\t\t'HSCHR6_1_CTG9'\t170815583\tStandard\n",
      "#\t\t'HSCHR6_1_CTG5'\t170815302\tStandard\n",
      "#\t\t'HSCHR6_8_CTG1'\t170813052\tStandard\n",
      "#\t\t'HSCHR6_1_CTG7'\t170812567\tStandard\n",
      "#\t\t'HG2128_PATCH'\t170811684\tStandard\n",
      "#\t\t'HSCHR6_1_CTG10'\t170810830\tStandard\n",
      "#\t\t'HSCHR6_1_CTG8'\t170810115\tStandard\n",
      "#\t\t'HSCHR6_1_CTG4'\t170808607\tStandard\n",
      "#\t\t'6'\t170805979\tStandard\n",
      "#\t\t'HSCHR6_MHC_DBB_CTG1'\t170783898\tStandard\n",
      "#\t\t'HG1651_PATCH'\t170779650\tStandard\n",
      "#\t\t'HSCHR6_MHC_QBL_CTG1'\t170734801\tStandard\n",
      "#\t\t'HSCHR6_MHC_COX_CTG1'\t170727598\tStandard\n",
      "#\t\t'HSCHR7_2_CTG6'\t159407160\tStandard\n",
      "#\t\t'HSCHR7_1_CTG6'\t159351981\tStandard\n",
      "#\t\t'HSCHR7_2_CTG7'\t159349352\tStandard\n",
      "#\t\t'HSCHR7_1_CTG1'\t159349278\tStandard\n",
      "#\t\t'HSCHR7_1_CTG4_4'\t159348604\tStandard\n",
      "#\t\t'HSCHR7_2_CTG1'\t159348294\tStandard\n",
      "#\t\t'HSCHR7_1_CTG7'\t159346281\tStandard\n",
      "#\t\t'7'\t159345973\tStandard\n",
      "#\t\t'HG2239_PATCH'\t159345319\tStandard\n",
      "#\t\t'HSCHR7_3_CTG6'\t159344441\tStandard\n",
      "#\t\t'HSCHR7_2_CTG4_4'\t159333789\tStandard\n",
      "#\t\t'HSCHRX_2_CTG12'\t156053172\tStandard\n",
      "#\t\t'HSCHRX_1_CTG3'\t156043585\tStandard\n",
      "#\t\t'X'\t156040895\tStandard\n",
      "#\t\t'HSCHRX_2_CTG3'\t156039875\tStandard\n",
      "#\t\t'HSCHR8_1_CTG6'\t145150400\tStandard\n",
      "#\t\t'HSCHR8_5_CTG7'\t145144538\tStandard\n",
      "#\t\t'HSCHR8_4_CTG7'\t145143933\tStandard\n",
      "#\t\t'HSCHR8_1_CTG7'\t145142033\tStandard\n",
      "#\t\t'HSCHR8_2_CTG7'\t145141868\tStandard\n",
      "#\t\t'HSCHR8_6_CTG7'\t145141837\tStandard\n",
      "#\t\t'HSCHR8_2_CTG1'\t145141567\tStandard\n",
      "#\t\t'HSCHR8_4_CTG1'\t145141010\tStandard\n",
      "#\t\t'8'\t145138636\tStandard\n",
      "#\t\t'HSCHR8_7_CTG1'\t145138198\tStandard\n",
      "#\t\t'HSCHR8_5_CTG1'\t145137619\tStandard\n",
      "#\t\t'HSCHR8_6_CTG1'\t145136257\tStandard\n",
      "#\t\t'HSCHR8_3_CTG1'\t145130612\tStandard\n",
      "#\t\t'HSCHR8_3_CTG7'\t145125643\tStandard\n",
      "#\t\t'HSCHR8_1_CTG1'\t145124952\tStandard\n",
      "#\t\t'HSCHR8_8_CTG1'\t145123326\tStandard\n",
      "#\t\t'HSCHR8_9_CTG1'\t144983589\tStandard\n",
      "#\t\t'HSCHR9_1_CTG1'\t138409529\tStandard\n",
      "#\t\t'HSCHR9_1_CTG4'\t138405815\tStandard\n",
      "#\t\t'HSCHR9_1_CTG5'\t138404305\tStandard\n",
      "#\t\t'HSCHR9_1_CTG2'\t138402913\tStandard\n",
      "#\t\t'HSCHR9_1_CTG3'\t138400797\tStandard\n",
      "#\t\t'HSCHR9_1_CTG6'\t138399532\tStandard\n",
      "#\t\t'HSCHR9_1_CTG7'\t138399111\tStandard\n",
      "#\t\t'HG2030_PATCH'\t138394865\tStandard\n",
      "#\t\t'9'\t138394717\tStandard\n",
      "#\t\t'HG151_NOVEL_TEST'\t135097696\tStandard\n",
      "#\t\t'HSCHR11_1_CTG5'\t135097255\tStandard\n",
      "#\t\t'HSCHR11_1_CTG1_2'\t135096748\tStandard\n",
      "#\t\t'HG142_HG150_NOVEL_TEST'\t135094228\tStandard\n",
      "#\t\t'HSCHR11_2_CTG1'\t135093444\tStandard\n",
      "#\t\t'HSCHR11_1_CTG1_1'\t135092355\tStandard\n",
      "#\t\t'HSCHR11_1_CTG6'\t135092332\tStandard\n",
      "#\t\t'HSCHR11_3_CTG1'\t135092180\tStandard\n",
      "#\t\t'HG2116_PATCH'\t135090698\tStandard\n",
      "#\t\t'HSCHR11_1_CTG3'\t135090275\tStandard\n",
      "#\t\t'HG107_PATCH'\t135088590\tStandard\n",
      "#\t\t'HSCHR11_1_CTG8'\t135088433\tStandard\n",
      "#\t\t'11'\t135086622\tStandard\n",
      "#\t\t'HG2217_PATCH'\t135086621\tStandard\n",
      "#\t\t'HSCHR11_1_CTG7'\t135085484\tStandard\n",
      "#\t\t'HSCHR11_2_CTG1_1'\t135083841\tStandard\n",
      "#\t\t'HSCHR11_1_CTG2'\t135083598\tStandard\n",
      "#\t\t'HSCHR10_1_CTG1'\t133844722\tStandard\n",
      "#\t\t'HSCHR10_1_CTG4'\t133815819\tStandard\n",
      "#\t\t'HSCHR10_1_CTG2'\t133813485\tStandard\n",
      "#\t\t'HSCHR10_1_CTG6'\t133801239\tStandard\n",
      "#\t\t'HG2191_PATCH'\t133799479\tStandard\n",
      "#\t\t'HG2244_HG2245_PATCH'\t133797425\tStandard\n",
      "#\t\t'10'\t133797422\tStandard\n",
      "#\t\t'HG2334_PATCH'\t133797389\tStandard\n",
      "#\t\t'HG2242_HG2243_PATCH'\t133797297\tStandard\n",
      "#\t\t'HG2241_PATCH'\t133792722\tStandard\n",
      "#\t\t'HSCHR10_1_CTG3'\t133783218\tStandard\n",
      "#\t\t'HSCHR12_2_CTG2'\t133307257\tStandard\n",
      "#\t\t'HSCHR12_7_CTG2_1'\t133289478\tStandard\n",
      "#\t\t'HSCHR12_1_CTG2_1'\t133284468\tStandard\n",
      "#\t\t'HSCHR12_2_CTG2_1'\t133282358\tStandard\n",
      "#\t\t'HSCHR12_2_CTG1'\t133282237\tStandard\n",
      "#\t\t'HSCHR12_3_CTG2_1'\t133281486\tStandard\n",
      "#\t\t'HSCHR12_1_CTG2'\t133281368\tStandard\n",
      "#\t\t'HSCHR12_4_CTG2_1'\t133281351\tStandard\n",
      "#\t\t'HSCHR12_5_CTG2_1'\t133279478\tStandard\n",
      "#\t\t'HSCHR12_1_CTG1'\t133279324\tStandard\n",
      "#\t\t'HSCHR12_4_CTG2'\t133277582\tStandard\n",
      "#\t\t'HSCHR12_6_CTG2_1'\t133275361\tStandard\n",
      "#\t\t'HG1362_PATCH'\t133275314\tStandard\n",
      "#\t\t'12'\t133275309\tStandard\n",
      "#\t\t'HSCHR12_5_CTG2'\t133275237\tStandard\n",
      "#\t\t'HG23_PATCH'\t133275184\tStandard\n",
      "#\t\t'HG2063_PATCH'\t133275140\tStandard\n",
      "#\t\t'HG2247_PATCH'\t133274000\tStandard\n",
      "#\t\t'HSCHR12_3_CTG2'\t133271566\tStandard\n",
      "#\t\t'HG2291_PATCH'\t114534236\tStandard\n",
      "#\t\t'HG2216_PATCH'\t114376428\tStandard\n",
      "#\t\t'HSCHR13_1_CTG7'\t114374491\tStandard\n",
      "#\t\t'HSCHR13_1_CTG3'\t114370226\tStandard\n",
      "#\t\t'HSCHR13_1_CTG8'\t114368996\tStandard\n",
      "#\t\t'HSCHR13_1_CTG4'\t114368332\tStandard\n",
      "#\t\t'HSCHR13_1_CTG2'\t114368315\tStandard\n",
      "#\t\t'HSCHR13_1_CTG6'\t114367169\tStandard\n",
      "#\t\t'HSCHR13_1_CTG5'\t114365350\tStandard\n",
      "#\t\t'13'\t114364328\tStandard\n",
      "#\t\t'HG2249_PATCH'\t114364211\tStandard\n",
      "#\t\t'HG2288_HG2289_PATCH'\t114360394\tStandard\n",
      "#\t\t'HSCHR13_1_CTG1'\t114348172\tStandard\n",
      "#\t\t'HSCHR14_3_CTG1'\t107067894\tStandard\n",
      "#\t\t'HSCHR14_2_CTG1'\t107050862\tStandard\n",
      "#\t\t'HSCHR14_1_CTG1'\t107047377\tStandard\n",
      "#\t\t'HSCHR14_7_CTG1'\t107045728\tStandard\n",
      "#\t\t'14'\t107043718\tStandard\n",
      "#\t\t'HSCHR15_4_CTG8'\t102071387\tStandard\n",
      "#\t\t'HSCHR15_5_CTG8'\t102070278\tStandard\n",
      "#\t\t'HSCHR15_2_CTG8'\t102003326\tStandard\n",
      "#\t\t'HSCHR15_1_CTG8'\t102002809\tStandard\n",
      "#\t\t'HSCHR15_3_CTG8'\t101999716\tStandard\n",
      "#\t\t'HSCHR15_1_CTG1'\t101998275\tStandard\n",
      "#\t\t'HSCHR15_2_CTG3'\t101995966\tStandard\n",
      "#\t\t'HSCHR15_1_CTG3'\t101994657\tStandard\n",
      "#\t\t'HSCHR15_3_CTG3'\t101993763\tStandard\n",
      "#\t\t'15'\t101991189\tStandard\n",
      "#\t\t'HSCHR15_6_CTG8'\t101521929\tStandard\n",
      "#\t\t'HSCHR16_1_CTG1'\t90426449\tStandard\n",
      "#\t\t'HSCHR16_3_CTG3_1'\t90346170\tStandard\n",
      "#\t\t'HSCHR16_5_CTG1'\t90344275\tStandard\n",
      "#\t\t'HSCHR16_2_CTG3_1'\t90343477\tStandard\n",
      "#\t\t'HSCHR16_4_CTG3_1'\t90343133\tStandard\n",
      "#\t\t'HSCHR16_CTG2'\t90341867\tStandard\n",
      "#\t\t'HSCHR16_4_CTG1'\t90340369\tStandard\n",
      "#\t\t'HSCHR16_3_CTG1'\t90340304\tStandard\n",
      "#\t\t'16'\t90338345\tStandard\n",
      "#\t\t'HSCHR16_1_CTG3_1'\t90335239\tStandard\n",
      "#\t\t'HSCHR17_1_CTG5'\t83552665\tStandard\n",
      "#\t\t'HSCHR17_10_CTG4'\t83359087\tStandard\n",
      "#\t\t'HSCHR17_1_CTG1'\t83290495\tStandard\n",
      "#\t\t'HSCHR17_3_CTG4'\t83269063\tStandard\n",
      "#\t\t'HSCHR17_1_CTG4'\t83264766\tStandard\n",
      "#\t\t'HSCHR17_9_CTG4'\t83263791\tStandard\n",
      "#\t\t'HSCHR17_2_CTG4'\t83263494\tStandard\n",
      "#\t\t'HSCHR17_2_CTG1'\t83259496\tStandard\n",
      "#\t\t'HSCHR17_8_CTG4'\t83258474\tStandard\n",
      "#\t\t'HSCHR17_3_CTG2'\t83258211\tStandard\n",
      "#\t\t'HSCHR17_4_CTG4'\t83257896\tStandard\n",
      "#\t\t'17'\t83257441\tStandard\n",
      "#\t\t'HSCHR17_1_CTG2'\t83256362\tStandard\n",
      "#\t\t'HSCHR17_1_CTG9'\t83255504\tStandard\n",
      "#\t\t'HSCHR17_5_CTG4'\t83247119\tStandard\n",
      "#\t\t'HSCHR17_6_CTG4'\t83246071\tStandard\n",
      "#\t\t'HSCHR17_2_CTG2'\t83160220\tStandard\n",
      "#\t\t'HSCHR17_2_CTG5'\t83086355\tStandard\n",
      "#\t\t'HSCHR17_7_CTG4'\t83058228\tStandard\n",
      "#\t\t'HSCHR18_2_CTG2'\t80442734\tStandard\n",
      "#\t\t'HSCHR18_1_CTG2'\t80386177\tStandard\n",
      "#\t\t'HSCHR18_1_CTG2_1'\t80382671\tStandard\n",
      "#\t\t'HSCHR18_1_CTG1_1'\t80380517\tStandard\n",
      "#\t\t'HSCHR18_2_CTG1_1'\t80379367\tStandard\n",
      "#\t\t'HSCHR18_3_CTG2_1'\t80379039\tStandard\n",
      "#\t\t'HSCHR18_5_CTG1_1'\t80378965\tStandard\n",
      "#\t\t'HSCHR18_2_CTG2_1'\t80378720\tStandard\n",
      "#\t\t'HSCHR18_4_CTG1_1'\t80378406\tStandard\n",
      "#\t\t'HG2213_PATCH'\t80377536\tStandard\n",
      "#\t\t'HSCHR18_ALT2_CTG2_1'\t80376883\tStandard\n",
      "#\t\t'HSCHR18_ALT21_CTG2_1'\t80376343\tStandard\n",
      "#\t\t'18'\t80373285\tStandard\n",
      "#\t\t'HSCHR20_1_CTG1'\t64450183\tStandard\n",
      "#\t\t'HSCHR20_1_CTG3'\t64447812\tStandard\n",
      "#\t\t'HSCHR20_1_CTG4'\t64446453\tStandard\n",
      "#\t\t'20'\t64444167\tStandard\n",
      "#\t\t'HSCHR20_1_CTG2'\t64441059\tStandard\n",
      "#\t\t'HSCHR19KIR_FH15_B_HAP_CTG3_1'\t58689284\tStandard\n",
      "#\t\t'HSCHR19KIR_FH05_B_HAP_CTG3_1'\t58689016\tStandard\n",
      "#\t\t'HSCHR19KIR_FH13_BA2_HAP_CTG3_1'\t58662330\tStandard\n",
      "#\t\t'HSCHR19KIR_RSH_BA2_HAP_CTG3_1'\t58662237\tStandard\n",
      "#\t\t'HSCHR19KIR_G248_BA2_HAP_CTG3_1'\t58661364\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_J_CTG3_1'\t58650772\tStandard\n",
      "#\t\t'HSCHR19KIR_FH08_BAX_HAP_CTG3_1'\t58647709\tStandard\n",
      "#\t\t'HSCHR19KIR_GRC212_AB_HAP_CTG3_1'\t58644315\tStandard\n",
      "#\t\t'HG2021_PATCH'\t58632883\tStandard\n",
      "#\t\t'HSCHR19KIR_LUCE_BDEL_HAP_CTG3_1'\t58631446\tStandard\n",
      "#\t\t'HSCHR19KIR_RP5_B_HAP_CTG3_1'\t58631402\tStandard\n",
      "#\t\t'HSCHR19KIR_T7526_BDEL_HAP_CTG3_1'\t58631206\tStandard\n",
      "#\t\t'HSCHR19_2_CTG2'\t58627095\tStandard\n",
      "#\t\t'HSCHR19_3_CTG3_1'\t58625958\tStandard\n",
      "#\t\t'HSCHR19_3_CTG2'\t58625823\tStandard\n",
      "#\t\t'HSCHR19_4_CTG3_1'\t58625731\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_S_CTG3_1'\t58625321\tStandard\n",
      "#\t\t'HSCHR19_1_CTG3_1'\t58624789\tStandard\n",
      "#\t\t'HSCHR19_1_CTG2'\t58624032\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_I_CTG3_1'\t58623235\tStandard\n",
      "#\t\t'HSCHR19_4_CTG2'\t58619803\tStandard\n",
      "#\t\t'HSCHR19_5_CTG2'\t58619095\tStandard\n",
      "#\t\t'HSCHR19_2_CTG3_1'\t58618601\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_AB_HAP_C_P_CTG3_1'\t58618008\tStandard\n",
      "#\t\t'HSCHR19KIR_FH08_A_HAP_CTG3_1'\t58617658\tStandard\n",
      "#\t\t'HG26_PATCH'\t58617652\tStandard\n",
      "#\t\t'HSCHR19KIR_G248_A_HAP_CTG3_1'\t58617652\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_AB_HAP_T_P_CTG3_1'\t58617636\tStandard\n",
      "#\t\t'HSCHR19KIR_LUCE_A_HAP_CTG3_1'\t58617634\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_A1_HAP_CTG3_1'\t58617625\tStandard\n",
      "#\t\t'19'\t58617616\tStandard\n",
      "#\t\t'HSCHR19KIR_RSH_A_HAP_CTG3_1'\t58617616\tStandard\n",
      "#\t\t'HSCHR19KIR_G085_A_HAP_CTG3_1'\t58617601\tStandard\n",
      "#\t\t'HSCHR19KIR_T7526_A_HAP_CTG3_1'\t58617601\tStandard\n",
      "#\t\t'HSCHR19KIR_FH06_A_HAP_CTG3_1'\t58617587\tStandard\n",
      "#\t\t'HSCHR19KIR_FH15_A_HAP_CTG3_1'\t58617473\tStandard\n",
      "#\t\t'HSCHR19KIR_FH05_A_HAP_CTG3_1'\t58616795\tStandard\n",
      "#\t\t'HSCHR19KIR_FH13_A_HAP_CTG3_1'\t58616725\tStandard\n",
      "#\t\t'HSCHR19KIR_FH06_BA1_HAP_CTG3_1'\t58603994\tStandard\n",
      "#\t\t'HSCHR19KIR_G085_BA1_HAP_CTG3_1'\t58603800\tStandard\n",
      "#\t\t'HSCHR19KIR_GRC212_BA1_HAP_CTG3_1'\t58602595\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_T_CTG3_1'\t58561614\tStandard\n",
      "#\t\t'HSCHR19LRC_COX1_CTG3_1'\t58546647\tStandard\n",
      "#\t\t'HSCHR19LRC_PGF1_CTG3_1'\t58546031\tStandard\n",
      "#\t\t'HSCHR19LRC_PGF2_CTG3_1'\t58355410\tStandard\n",
      "#\t\t'HSCHR19LRC_COX2_CTG3_1'\t58288451\tStandard\n",
      "#\t\t'HG2062_PATCH'\t57262172\tStandard\n",
      "#\t\t'Y'\t57227415\tStandard\n",
      "#\t\t'HSCHR22_1_CTG7'\t50872732\tStandard\n",
      "#\t\t'HSCHR22_7_CTG1'\t50845852\tStandard\n",
      "#\t\t'HSCHR22_1_CTG6'\t50842710\tStandard\n",
      "#\t\t'HSCHR22_5_CTG1'\t50832153\tStandard\n",
      "#\t\t'HSCHR22_8_CTG1'\t50832130\tStandard\n",
      "#\t\t'HSCHR22_2_CTG1'\t50830603\tStandard\n",
      "#\t\t'HSCHR22_1_CTG3'\t50827573\tStandard\n",
      "#\t\t'HSCHR22_1_CTG4'\t50824699\tStandard\n",
      "#\t\t'HSCHR22_1_CTG5'\t50821308\tStandard\n",
      "#\t\t'HG1311_PATCH'\t50818486\tStandard\n",
      "#\t\t'22'\t50818468\tStandard\n",
      "#\t\t'HSCHR22_3_CTG1'\t50818468\tStandard\n",
      "#\t\t'HSCHR22_6_CTG1'\t50815919\tStandard\n",
      "#\t\t'HSCHR22_1_CTG1'\t50806372\tStandard\n",
      "#\t\t'HSCHR22_4_CTG1'\t50806321\tStandard\n",
      "#\t\t'HSCHR22_1_CTG2'\t50788525\tStandard\n",
      "#\t\t'HSCHR21_1_CTG1_1'\t46722569\tStandard\n",
      "#\t\t'HSCHR21_4_CTG1_1'\t46719541\tStandard\n",
      "#\t\t'HSCHR21_2_CTG1_1'\t46716685\tStandard\n",
      "#\t\t'HSCHR21_3_CTG1_1'\t46715777\tStandard\n",
      "#\t\t'HSCHR21_6_CTG1_1'\t46713516\tStandard\n",
      "#\t\t'HSCHR21_8_CTG1_1'\t46713185\tStandard\n",
      "#\t\t'HSCHR21_5_CTG2'\t46712294\tStandard\n",
      "#\t\t'21'\t46709983\tStandard\n",
      "#\t\t'KI270728.1'\t1872759\tStandard\n",
      "#\t\t'KI270727.1'\t448248\tStandard\n",
      "#\t\t'KI270442.1'\t392061\tStandard\n",
      "#\t\t'KI270729.1'\t280839\tStandard\n",
      "#\t\t'GL000225.1'\t211173\tStandard\n",
      "#\t\t'KI270743.1'\t210658\tStandard\n",
      "#\t\t'GL000008.2'\t209709\tStandard\n",
      "#\t\t'GL000009.2'\t201709\tStandard\n",
      "#\t\t'KI270747.1'\t198735\tStandard\n",
      "#\t\t'KI270722.1'\t194050\tStandard\n",
      "#\t\t'GL000194.1'\t191469\tStandard\n",
      "#\t\t'KI270742.1'\t186739\tStandard\n",
      "#\t\t'GL000205.2'\t185591\tStandard\n",
      "#\t\t'GL000195.1'\t182896\tStandard\n",
      "#\t\t'KI270736.1'\t181920\tStandard\n",
      "#\t\t'KI270733.1'\t179772\tStandard\n",
      "#\t\t'GL000224.1'\t179693\tStandard\n",
      "#\t\t'GL000219.1'\t179198\tStandard\n",
      "#\t\t'KI270719.1'\t176845\tStandard\n",
      "#\t\t'GL000216.2'\t176608\tStandard\n",
      "#\t\t'KI270712.1'\t176043\tStandard\n",
      "#\t\t'KI270706.1'\t175055\tStandard\n",
      "#\t\t'KI270725.1'\t172810\tStandard\n",
      "#\t\t'KI270744.1'\t168472\tStandard\n",
      "#\t\t'KI270734.1'\t165050\tStandard\n",
      "#\t\t'GL000213.1'\t164239\tStandard\n",
      "#\t\t'GL000220.1'\t161802\tStandard\n",
      "#\t\t'KI270715.1'\t161471\tStandard\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "#\t\t'GL000218.1'\t161147\tStandard\n",
      "#\t\t'KI270749.1'\t158759\tStandard\n",
      "#\t\t'KI270741.1'\t157432\tStandard\n",
      "#\t\t'GL000221.1'\t155397\tStandard\n",
      "#\t\t'KI270716.1'\t153799\tStandard\n",
      "#\t\t'KI270731.1'\t150754\tStandard\n",
      "#\t\t'KI270751.1'\t150742\tStandard\n",
      "#\t\t'KI270750.1'\t148850\tStandard\n",
      "#\t\t'KI270519.1'\t138126\tStandard\n",
      "#\t\t'GL000214.1'\t137718\tStandard\n",
      "#\t\t'KI270708.1'\t127682\tStandard\n",
      "#\t\t'KI270730.1'\t112551\tStandard\n",
      "#\t\t'KI270438.1'\t112505\tStandard\n",
      "#\t\t'KI270737.1'\t103838\tStandard\n",
      "#\t\t'KI270721.1'\t100316\tStandard\n",
      "#\t\t'KI270738.1'\t99375\tStandard\n",
      "#\t\t'KI270748.1'\t93321\tStandard\n",
      "#\t\t'KI270435.1'\t92983\tStandard\n",
      "#\t\t'GL000208.1'\t92689\tStandard\n",
      "#\t\t'KI270538.1'\t91309\tStandard\n",
      "#\t\t'KI270756.1'\t79590\tStandard\n",
      "#\t\t'KI270739.1'\t73985\tStandard\n",
      "#\t\t'KI270757.1'\t71251\tStandard\n",
      "#\t\t'KI270709.1'\t66860\tStandard\n",
      "#\t\t'KI270746.1'\t66486\tStandard\n",
      "#\t\t'KI270753.1'\t62944\tStandard\n",
      "#\t\t'KI270589.1'\t44474\tStandard\n",
      "#\t\t'KI270726.1'\t43739\tStandard\n",
      "#\t\t'KI270735.1'\t42811\tStandard\n",
      "#\t\t'KI270711.1'\t42210\tStandard\n",
      "#\t\t'KI270745.1'\t41891\tStandard\n",
      "#\t\t'KI270714.1'\t41717\tStandard\n",
      "#\t\t'KI270732.1'\t41543\tStandard\n",
      "#\t\t'KI270713.1'\t40745\tStandard\n",
      "#\t\t'KI270754.1'\t40191\tStandard\n",
      "#\t\t'KI270710.1'\t40176\tStandard\n",
      "#\t\t'KI270717.1'\t40062\tStandard\n",
      "#\t\t'KI270724.1'\t39555\tStandard\n",
      "#\t\t'KI270720.1'\t39050\tStandard\n",
      "#\t\t'KI270723.1'\t38115\tStandard\n",
      "#\t\t'KI270718.1'\t38054\tStandard\n",
      "#\t\t'KI270317.1'\t37690\tStandard\n",
      "#\t\t'KI270740.1'\t37240\tStandard\n",
      "#\t\t'KI270755.1'\t36723\tStandard\n",
      "#\t\t'KI270707.1'\t32032\tStandard\n",
      "#\t\t'KI270579.1'\t31033\tStandard\n",
      "#\t\t'KI270752.1'\t27745\tStandard\n",
      "#\t\t'KI270512.1'\t22689\tStandard\n",
      "#\t\t'KI270322.1'\t21476\tStandard\n",
      "#\t\t'MT'\t16569\tStandard\n",
      "#\t\t'GL000226.1'\t15008\tStandard\n",
      "#\t\t'KI270311.1'\t12399\tStandard\n",
      "#\t\t'KI270366.1'\t8320\tStandard\n",
      "#\t\t'KI270511.1'\t8127\tStandard\n",
      "#\t\t'KI270448.1'\t7992\tStandard\n",
      "#\t\t'KI270521.1'\t7642\tStandard\n",
      "#\t\t'KI270581.1'\t7046\tStandard\n",
      "#\t\t'KI270582.1'\t6504\tStandard\n",
      "#\t\t'KI270515.1'\t6361\tStandard\n",
      "#\t\t'KI270588.1'\t6158\tStandard\n",
      "#\t\t'KI270591.1'\t5796\tStandard\n",
      "#\t\t'KI270522.1'\t5674\tStandard\n",
      "#\t\t'KI270507.1'\t5353\tStandard\n",
      "#\t\t'KI270590.1'\t4685\tStandard\n",
      "#\t\t'KI270584.1'\t4513\tStandard\n",
      "#\t\t'KI270320.1'\t4416\tStandard\n",
      "#\t\t'KI270382.1'\t4215\tStandard\n",
      "#\t\t'KI270468.1'\t4055\tStandard\n",
      "#\t\t'KI270467.1'\t3920\tStandard\n",
      "#\t\t'KI270362.1'\t3530\tStandard\n",
      "#\t\t'KI270517.1'\t3253\tStandard\n",
      "#\t\t'KI270593.1'\t3041\tStandard\n",
      "#\t\t'KI270528.1'\t2983\tStandard\n",
      "#\t\t'KI270587.1'\t2969\tStandard\n",
      "#\t\t'KI270364.1'\t2855\tStandard\n",
      "#\t\t'KI270371.1'\t2805\tStandard\n",
      "#\t\t'KI270333.1'\t2699\tStandard\n",
      "#\t\t'KI270374.1'\t2656\tStandard\n",
      "#\t\t'KI270411.1'\t2646\tStandard\n",
      "#\t\t'KI270414.1'\t2489\tStandard\n",
      "#\t\t'KI270510.1'\t2415\tStandard\n",
      "#\t\t'KI270390.1'\t2387\tStandard\n",
      "#\t\t'KI270375.1'\t2378\tStandard\n",
      "#\t\t'KI270420.1'\t2321\tStandard\n",
      "#\t\t'KI270509.1'\t2318\tStandard\n",
      "#\t\t'KI270315.1'\t2276\tStandard\n",
      "#\t\t'KI270302.1'\t2274\tStandard\n",
      "#\t\t'KI270518.1'\t2186\tStandard\n",
      "#\t\t'KI270530.1'\t2168\tStandard\n",
      "#\t\t'KI270304.1'\t2165\tStandard\n",
      "#\t\t'KI270418.1'\t2145\tStandard\n",
      "#\t\t'KI270424.1'\t2140\tStandard\n",
      "#\t\t'KI270417.1'\t2043\tStandard\n",
      "#\t\t'KI270508.1'\t1951\tStandard\n",
      "#\t\t'KI270303.1'\t1942\tStandard\n",
      "#\t\t'KI270381.1'\t1930\tStandard\n",
      "#\t\t'KI270529.1'\t1899\tStandard\n",
      "#\t\t'KI270425.1'\t1884\tStandard\n",
      "#\t\t'KI270396.1'\t1880\tStandard\n",
      "#\t\t'KI270363.1'\t1803\tStandard\n",
      "#\t\t'KI270386.1'\t1788\tStandard\n",
      "#\t\t'KI270465.1'\t1774\tStandard\n",
      "#\t\t'KI270383.1'\t1750\tStandard\n",
      "#\t\t'KI270384.1'\t1658\tStandard\n",
      "#\t\t'KI270330.1'\t1652\tStandard\n",
      "#\t\t'KI270372.1'\t1650\tStandard\n",
      "#\t\t'KI270548.1'\t1599\tStandard\n",
      "#\t\t'KI270580.1'\t1553\tStandard\n",
      "#\t\t'KI270387.1'\t1537\tStandard\n",
      "#\t\t'KI270391.1'\t1484\tStandard\n",
      "#\t\t'KI270305.1'\t1472\tStandard\n",
      "#\t\t'KI270373.1'\t1451\tStandard\n",
      "#\t\t'KI270422.1'\t1445\tStandard\n",
      "#\t\t'KI270316.1'\t1444\tStandard\n",
      "#\t\t'KI270338.1'\t1428\tStandard\n",
      "#\t\t'KI270340.1'\t1428\tStandard\n",
      "#\t\t'KI270583.1'\t1400\tStandard\n",
      "#\t\t'KI270334.1'\t1368\tStandard\n",
      "#\t\t'KI270429.1'\t1361\tStandard\n",
      "#\t\t'KI270393.1'\t1308\tStandard\n",
      "#\t\t'KI270516.1'\t1300\tStandard\n",
      "#\t\t'KI270389.1'\t1298\tStandard\n",
      "#\t\t'KI270466.1'\t1233\tStandard\n",
      "#\t\t'KI270388.1'\t1216\tStandard\n",
      "#\t\t'KI270544.1'\t1202\tStandard\n",
      "#\t\t'KI270310.1'\t1201\tStandard\n",
      "#\t\t'KI270412.1'\t1179\tStandard\n",
      "#\t\t'KI270395.1'\t1143\tStandard\n",
      "#\t\t'KI270376.1'\t1136\tStandard\n",
      "#\t\t'KI270337.1'\t1121\tStandard\n",
      "#\t\t'KI270335.1'\t1048\tStandard\n",
      "#\t\t'KI270378.1'\t1048\tStandard\n",
      "#\t\t'KI270379.1'\t1045\tStandard\n",
      "#\t\t'KI270329.1'\t1040\tStandard\n",
      "#\t\t'KI270419.1'\t1029\tStandard\n",
      "#\t\t'KI270336.1'\t1026\tStandard\n",
      "#\t\t'KI270312.1'\t998\tStandard\n",
      "#\t\t'KI270539.1'\t993\tStandard\n",
      "#\t\t'KI270385.1'\t990\tStandard\n",
      "#\t\t'KI270423.1'\t981\tStandard\n",
      "#\t\t'KI270392.1'\t971\tStandard\n",
      "#\t\t'KI270394.1'\t970\tStandard\n",
      "#-----------------------------------------------\n",
      "\n",
      "00:00:32\tPredicting variants\n",
      "00:00:32\tLoading sequences for chromosome '3' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.3.bin'\n",
      "00:00:34\tBuilding sequence tree for chromosome '3'\n",
      "00:00:34\tDone. Loaded 1592 sequences.\n",
      "00:00:35\tLoading sequences for chromosome '6' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.6.bin'\n",
      "00:00:37\tBuilding sequence tree for chromosome '6'\n",
      "00:00:37\tDone. Loaded 1815 sequences.\n",
      "00:00:38\tLoading sequences for chromosome '15' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.15.bin'\n",
      "00:00:39\tBuilding sequence tree for chromosome '15'\n",
      "00:00:39\tDone. Loaded 1001 sequences.\n",
      "00:00:39\tLoading sequences for chromosome '19' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.19.bin'\n",
      "00:00:40\tBuilding sequence tree for chromosome '19'\n",
      "00:00:40\tDone. Loaded 1587 sequences.\n",
      "00:00:40\tLoading sequences for chromosome '22' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.22.bin'\n",
      "00:00:41\tBuilding sequence tree for chromosome '22'\n",
      "00:00:41\tDone. Loaded 774 sequences.\n",
      "00:00:41\t\t10000 variants (1093 variants per second), 9981 VCF entries\n",
      "00:00:41\tLoading sequences for chromosome '1' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.1.bin'\n",
      "00:00:44\tBuilding sequence tree for chromosome '1'\n",
      "00:00:44\tDone. Loaded 3134 sequences.\n",
      "00:00:47\tLoading sequences for chromosome '2' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.2.bin'\n",
      "00:00:49\tBuilding sequence tree for chromosome '2'\n",
      "00:00:49\tDone. Loaded 2321 sequences.\n",
      "00:00:50\tLoading sequences for chromosome '4' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.4.bin'\n",
      "00:00:52\tBuilding sequence tree for chromosome '4'\n",
      "00:00:52\tDone. Loaded 1617 sequences.\n",
      "00:00:52\tLoading sequences for chromosome '5' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.5.bin'\n",
      "00:00:54\tBuilding sequence tree for chromosome '5'\n",
      "00:00:54\tDone. Loaded 1654 sequences.\n",
      "00:00:54\t\t20000 variants (904 variants per second), 19974 VCF entries\n",
      "00:00:54\tLoading sequences for chromosome '7' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.7.bin'\n",
      "00:00:56\tBuilding sequence tree for chromosome '7'\n",
      "00:00:56\tDone. Loaded 1692 sequences.\n",
      "00:00:57\tLoading sequences for chromosome '8' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.8.bin'\n",
      "00:00:58\tBuilding sequence tree for chromosome '8'\n",
      "00:00:58\tDone. Loaded 1377 sequences.\n",
      "00:00:59\tLoading sequences for chromosome '9' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.9.bin'\n",
      "00:01:01\tBuilding sequence tree for chromosome '9'\n",
      "00:01:01\tDone. Loaded 1414 sequences.\n",
      "00:01:01\tLoading sequences for chromosome '10' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.10.bin'\n",
      "00:01:02\tBuilding sequence tree for chromosome '10'\n",
      "00:01:02\tDone. Loaded 1328 sequences.\n",
      "00:01:04\tLoading sequences for chromosome '11' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.11.bin'\n",
      "00:01:05\tBuilding sequence tree for chromosome '11'\n",
      "00:01:05\tDone. Loaded 1874 sequences.\n",
      "00:01:06\t\t30000 variants (897 variants per second), 29955 VCF entries\n",
      "00:01:06\tLoading sequences for chromosome '12' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.12.bin'\n",
      "00:01:07\tBuilding sequence tree for chromosome '12'\n",
      "00:01:07\tDone. Loaded 1521 sequences.\n",
      "00:01:07\tLoading sequences for chromosome '13' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.13.bin'\n",
      "00:01:08\tBuilding sequence tree for chromosome '13'\n",
      "00:01:08\tDone. Loaded 814 sequences.\n",
      "00:01:08\tLoading sequences for chromosome '14' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.14.bin'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "00:01:09\tBuilding sequence tree for chromosome '14'\n",
      "00:01:09\tDone. Loaded 1181 sequences.\n",
      "00:01:10\tLoading sequences for chromosome '16' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.16.bin'\n",
      "00:01:11\tBuilding sequence tree for chromosome '16'\n",
      "00:01:11\tDone. Loaded 1228 sequences.\n",
      "00:01:11\tLoading sequences for chromosome '17' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.17.bin'\n",
      "00:01:12\tBuilding sequence tree for chromosome '17'\n",
      "00:01:12\tDone. Loaded 1468 sequences.\n",
      "00:01:12\tLoading sequences for chromosome '18' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.18.bin'\n",
      "00:01:13\tBuilding sequence tree for chromosome '18'\n",
      "00:01:13\tDone. Loaded 632 sequences.\n",
      "00:01:13\tLoading sequences for chromosome '20' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.20.bin'\n",
      "00:01:14\tBuilding sequence tree for chromosome '20'\n",
      "00:01:14\tDone. Loaded 865 sequences.\n",
      "00:01:14\t\t40000 variants (956 variants per second), 39942 VCF entries\n",
      "00:01:14\tLoading sequences for chromosome '21' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.21.bin'\n",
      "00:01:14\tBuilding sequence tree for chromosome '21'\n",
      "00:01:14\tDone. Loaded 523 sequences.\n",
      "00:01:14\tLoading sequences for chromosome 'X' from file '/home/kwat/miniconda3/share/snpeff-4.3.1t-2/./data/GRCh38.86/sequence.X.bin'\n",
      "00:01:16\tBuilding sequence tree for chromosome 'X'\n",
      "00:01:16\tDone. Loaded 1694 sequences.\n",
      "\n",
      "WARNINGS: Some warning were detected\n",
      "Warning type\tNumber of warnings\n",
      "INFO_REALIGN_3_PRIME\t2231\n",
      "WARNING_TRANSCRIPT_INCOMPLETE\t11874\n",
      "WARNING_TRANSCRIPT_INCOMPLETE&INFO_REALIGN_3_PRIME\t158\n",
      "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t189\n",
      "WARNING_TRANSCRIPT_NO_START_CODON\t9832\n",
      "WARNING_TRANSCRIPT_NO_START_CODON&INFO_REALIGN_3_PRIME\t200\n",
      "WARNING_TRANSCRIPT_NO_STOP_CODON\t6332\n",
      "WARNING_TRANSCRIPT_NO_STOP_CODON&INFO_REALIGN_3_PRIME\t45\n",
      "\n",
      "\n",
      "00:01:16\tCreating summary file: /home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/stats.csv\n",
      "00:01:16\tCreating summary file: /home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/stats.html\n",
      "00:01:16\tCreating genes file: /home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/stats.genes.txt\n",
      "00:01:18\tdone.\n",
      "00:01:18\tDone.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "`\u001b[4mtabix\u001b[24m \u001b[4m/home/kwat/garden/sample/process_germ_dna/find_variant/snpeff/snpeff.vcf.gz\u001b[24m`\n",
      "(2019-06-24T21:31:07.803) Done in 7 minutes, 279 milliseconds.\n"
     ]
    }
   ],
   "source": [
    "const find_mutation_dir = joinpath(\n",
    "    output_dir,\n",
    "    \"find_variant\",\n",
    ")\n",
    "\n",
    "Kraft.find_variant(\n",
    "    germ_bam,\n",
    "    nothing,\n",
    "    path_dict[\"dna.fa.bgz\"],\n",
    "    path_dict[\"chromosome.bed.gz\"],\n",
    "    is_targeted,\n",
    "    path_dict[\"chrn_n.tsv\"],\n",
    "    find_mutation_dir,\n",
    "    n_job,\n",
    "    gb_memory,\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "@webio": {
   "lastCommId": null,
   "lastKernelId": null
  },
  "kernelspec": {
   "display_name": "Julia 1.0.3",
   "language": "julia",
   "name": "julia-1.0"
  },
  "language_info": {
   "file_extension": ".jl",
   "mimetype": "application/julia",
   "name": "julia",
   "version": "1.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
